features/step_definitions/authentication_steps.rb

```diff
@@ -0,0 +1,27 @@
+Given(/^I have an account on the site$/) do
+  FactoryGirl.create(:user, email: "user@example.com", password: "password")
+end
+
+Given(/^My account is confirmed$/) do
+  User.last.confirm
```

Comment:
I can see how that would be less repetitive, but also less reusable. Since these tests shouldn't be changing I can look into setting up an instance variable.